src/main/java/core/basesyntax/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategy(Map<FruitTransaction.Operation, OperationHandler>
+                                     operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
+    }

```

Comment:
I'd recommend creating and initializing this map somewhere outside and pass it to the constructor of this class (see previous comment in `main` method).